src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +590,32 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefReadonlyParameters, diagnostics);
+
+                        if (seenReadonly)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (seenOut)
+                        {
+                            addERR_BadParameterModifiers(diagnostics, modifier, SyntaxKind.OutKeyword);
+                        }
+                        else if (seenIn)
+                        {
+                            addERR_BadParameterModifiers(diagnostics, modifier, SyntaxKind.InKeyword);
+                        }
+                        else if (seenParams)
+                        {
+                            addERR_ParamsCantBeWithModifier(diagnostics, modifier, SyntaxKind.ReadOnlyKeyword);
+                        }
```

Comment:
It looks like these checks are essentially duplicates of the checks performed for a `ref` modifier, and below we report an error if previous modifier is not `ref`. Therefore, consider simplifying the logic and removing these checks, they are going to produce more "noise". At the same time, I think it is more important to report  ```ERR_RefReadOnlyWrongOrdering``` rather than the "noise".